### ISSUE TITLE
Add game delegate

### ISSUE THREAD

**Issue by richelbilderbeek** [richelbilderbeek/djog_unos_2018#29]
**Is your feature request related to a problem? Please describe.**

We need a delegate, which is a class that 'can do something for you'. Examples are:

 * Close the game after 100 frames (for a minimal test)
 * Mash random buttons and mouse click for 10K frames (for rough testing)

**Describe the solution you'd like**

## 1. Create a class `game_delegate`:

```
class game_delegate
{
  public:
  void exec(game& g)
  {
    if (g.get_n_frames() == 100)
    {
      g.close();
    }
  }
};
```

## 2. Let `game` store a `game_delegate`

```c++
class game
{
  //...
  private:
  game_delegate m_game_delegate;
};
```

And in the member function that displays a frame:

```
m_gate_delegate.exec(*this);
```

**Describe alternatives you've considered**

None.

**Additional context**

 * The must also be a delegate that does nothing. This one is used in normal play
 * The delegate will handle more events in the future

**Comment by richelbilderbeek** [comment]
I suggest to let @annehinrichs22 or @Joshua260403 do this. Sure I will help. The one who assigns himself first gets the job :+1:

**Comment by richelbilderbeek** [comment]
As @Joshua260403 signed up for #31 , I assign @annehinrichs22 to this one. I think @annehinrichs22 will like this! If disagree, let me know.

**Comment by richelbilderbeek** [comment]
Will do so Friday/tomorrow, as it will also fix #75
